File: bubble_sort.py
import platform
import time
from memory_profiler import memory_usage

def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

def main():
    # Lendo os números do arquivo
    with open("arq.txt", "r") as file:
        numbers = [int(line.strip()) for line in file]

    # Informações do sistema
    print("Linguagem: Python")
    print("Versão:", platform.python_version())
    print("Sistema Operacional:", platform.system())
    print("Arquitetura:", platform.architecture()[0])

    # Monitoramento de desempenho
    start_time = time.time()
    memory_used = memory_usage((bubble_sort, (numbers,)), interval=0.1)[0] * 1024  

    end_time = time.time()
    elapsed_time = (end_time - start_time) * 1000  

    # Gravando o resultado no arquivo
    with open("arq-saida.txt", "w") as file:
        for num in numbers:
            file.write(f"{num}\n")
    
    print(f"Tempo (ms): {elapsed_time:.2f}")
    print(f"Memória utilizada (KB): {memory_used:.2f}")

if __name__ == "__main__":
    main()

File: bubble_sort.js
const fs = require('fs');
const os = require('os');
const process = require('process');

function bubbleSort(arr) {
  const n = arr.length;
  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n - i - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]]; 
      }
    }
  } 
}

function main() {
  // Lendo os números do arquivo
  fs.readFile('arq.txt', 'utf8', (err, data) => {
    if (err) {
      console.error('Erro ao abrir o arquivo', err);
      return;
    }

    const numbers = data.split('\n').map(line => parseInt(line.trim(), 10)).filter(Boolean);

    // Informações do sistema
    console.log("Linguagem: JavaScript (Node.js)");
    console.log("Versão:", process.version);
    console.log("Sistema Operacional:", os.platform());
    console.log("Arquitetura:", os.arch());

    // Monitoramento de desempenho
    const startMemory = process.memoryUsage().rss; 
    const startTime = process.hrtime(); 

    // Ordenando com Bubble Sort
    bubbleSort(numbers);

    const [seconds, nanoseconds] = process.hrtime(startTime);
    const elapsedTime = (seconds * 1000) + (nanoseconds / 1000000); 

    const endMemory = process.memoryUsage().rss; 
    // Gravando o resultado no arquivo
    fs.writeFile('arq-saida.txt', numbers.join('\n'), 'utf8', (err) => {
      if (err) {
        console.error('Erro ao gravar o arquivo', err);
        return;
      }

      const memoryUsed = (endMemory - startMemory) / 1024;

      console.log(`Tempo (ms): ${elapsedTime.toFixed(2)}`);
      console.log(`Memória utilizada (KB): ${memoryUsed.toFixed(2)}`);
    });
  });
}

main();
